### PR TITLE
feat(core): `Fn::ToJsonString` and `Fn::Length` intrinsic functions

### PR DESCRIPTION
**Pull request by jogold**
Add support for `Fn::ToJsonString` and `Fn::Length`.

The `AWS::LanguageExtensions` transform is automatically added.

See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ToJsonString.html
See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-length.html
See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-transform.html

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/core/lib/cfn-fn.ts [hunk 1]
```diff
@@ -412,6 +413,37 @@ export class Fn {
...
+    return new FnToJsonString(object).toString();
+  }
+
+  /**
+   * The intrinsic function `Fn::Length` returns the number of elements within an array
+   * or an intrinsic function that returns an array.
+   *
+   * @param array The array you want to return the number of elements from
+   */
+  public static len(array: any): number {
```
Comment (jogold) [inline packages/@aws-cdk/core/lib/cfn-fn.ts:436]: cannot do `public static length` here.

```
Static property 'length' conflicts with built-in property 'Function.length' of constructor function 'Fn'.ts(2699)
```

**Comment by jogold** [comment]
Needs `pr-linter/exempt-integ-test`

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Can you explain why you think that integ tests aren't needed here? I think it would probably be good to add that test coverage.

**Comment by jogold** [comment]
> Can you explain why you think that integ tests aren't needed here? I think it would probably be good to add that test coverage.

There are currently no integ tests in `core`, and I believe this would create a circular dependency `@aws-cdk/core > @aws-cdk/integ-tests > @aws-cdk/core`. Also I'm not sure it would bring much added value for a CF syntax.

**Comment by TheRealAmazonKendra** [comment]
> > Can you explain why you think that integ tests aren't needed here? I think it would probably be good to add that test coverage.
> 
> There are currently no integ tests in `core`, and I believe this would create a circular dependency `@aws-cdk/core > @aws-cdk/integ-tests > @aws-cdk/core`. Also I'm not sure it would bring much added value for a CF syntax.

Hmm. I don't like that we don't have any tests for core, but adding them would very much be out of scope for this PR.

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**